### PR TITLE
Adjust which playbook is run depending on which group was changed

### PR DESCRIPTION
**Pull request by pypingou**

### CODE REVIEW

**Comment by pypingou** [comment]
This hasn't had much testing :(

**Comment by ralphbean** [comment]
I don't understand.  What is the difference between the two playbooks?

Also, will this fail on some of the message types?  Do any of them have an odd ``group`` field?  See https://apps.fedoraproject.org/datagrepper/raw?topic=org.fedoraproject.prod.fas.role.update&topic=org.fedoraproject.prod.fas.group.member.sponsor&topic=org.fedoraproject.prod.fas.group.member.remove&topic=org.fedoraproject.prod.fas.user.update for a query.

**Comment by pypingou** [comment]
> I don't understand. What is the difference between the two playbooks?

``run_fasClient.yml`` runs on all our hosts but some that do not have fasClient, while ``run_fasClient_simple.yml`` only runs on some of the hosts (fedorapeople, fedorahosted, bastion, pkgs)

File: fedmsg_fasclient.py [hunk 1]
```diff
@@ -89,8 +89,18 @@ def delayed_consume():
     def action(self, messages):
         self.log.debug("Acting on %s" % pprint.pformat(messages))
 
+        playbook = 'run_fasClient_simple.yml'
+        groups = [
+            msg['msg']['group']
+            for msg in messages
+            if 'sysadmin-' in msg['msg'].get('group', None)
```
Comment (ralphbean) [inline fedmsg_fasclient.py]: Perhaps just ``'sysadmin'`` instead of ``'sysadmin-'``?

**Comment by ralphbean** [comment]
So this has us doing simple by default.  But the more complete one when the change is with respect to a group.

This deserves an inline comment to explain.  Why is it more complex when there's a group involved?  What kind of change happened that deserves only a simple run when there is no group involved.

Is it that the simple runs take place only when a public SSH key changes?

What about if a sysadmin with shell access changes their SSH key?  That info won't get propagated to all the nodes.

**Comment by pypingou** [comment]
- fas.group.member.sponsor has a group
- fas.user.update has no group
- fas.group.member.remove has a group
- fas.role.update has a group

So fas.user.update is the only one giving us problem :-/

**Comment by pypingou** [comment]
> What about if a sysadmin with shell access changes their SSH key? That info won't get propagated to all the nodes.

Looking into the data for my previous reply I was thinking about this and maybe we should look into the message for ``fas.user.update`` and use the global playbook if the user updated their ssh key.

Thoughts?

**Comment by pypingou** [comment]
>  Why is it more complex when there's a group involved? What kind of change happened that deserves only a simple run when there is no group involved.

It's not about having groups involved, it's about which groups are involved :)

File: fedmsg_fasclient.py (hunk 1 above)
Comment (pypingou) [inline fedmsg_fasclient.py]: I was wondering if there are groups having ``sysadmin`` in their name but not related to us (I'm thinking a ``foosysadmingit`` or so.

File: fedmsg_fasclient.py [hunk 2]
```diff
@@ -89,8 +89,32 @@ def delayed_consume():
...
+        # - ``run_fasClient.yml``        runs fasClient on all our hosts
+        # So by default we run on the subset and only run on all the hosts if
+        # the message concerned a group involving sysadmins or an user that
+        # changed their ssh key.
+        playbook = 'run_fasClient_simple.yml'
+        whole_playbook = False
+        for msg in messages:
+            # If one of the groups includes a sysadmin-related group, run the
+            # complete playbook
+            if 'sysadmin-' in msg['msg'].get('group', None) \
```
Comment (ralphbean) [inline fedmsg_fasclient.py]: Maybe leave the ``'-'`` off the end of ``'sysadmin-'`` so we can catch the "sysadmin" group itself?

**Comment by ralphbean** [comment]
Nice!  :+1:

**Pull request merged**